src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -70,56 +70,100 @@ export default class GroupComparisonStore {
     @observable private _currentTabId:GroupComparisonTab|undefined = undefined;
     @observable private _overlapStrategy:OverlapStrategy = OverlapStrategy.EXCLUDE;
     @observable private sessionId:string;
-    @observable dragNameOrder:string[]|undefined = undefined;
-    private _unsavedGroups = observable.shallowArray<SessionGroupData>([]);
-    private deletedGroupNames = observable.shallowMap<string>();
+    @observable public loading = false;
 
     constructor(sessionId:string, private appStore:AppStore) {
         this.sessionId = sessionId;
     }
 
-    @action public deleteGroup(name:string) {
-        this.deletedGroupNames.set(name, name);
+    @computed get groupOrder() {
+        const param = ((window as any).routingStore.location.query as GroupComparisonURLQuery).groupOrder;
+        if (param) {
+            return JSON.parse(param);
+        } else {
+            return undefined;
+        }
     }
 
-    @computed get existsDeletedGroup() {
-        return this.deletedGroupNames.keys().length > 0;
+    @action public updateGroupOrder(oldIndex:number, newIndex:number) {
+        let groupOrder = this.groupOrder;
+        if (!groupOrder) {
+            groupOrder = this._originalGroups.result!.map(g=>g.name);
+        }
+        groupOrder = groupOrder.slice();
+        const poppedUid = groupOrder.splice(oldIndex, 1)[0];
+        groupOrder.splice(newIndex, 0, poppedUid);
+
+        (window as any).routingStore.updateRoute({ groupOrder: JSON.stringify(groupOrder) } as Partial<GroupComparisonURLQuery>);
```

Comment:
may be join by `,`

Reply:
a group name could have a comma in it

Reply:
lets pass in routingStore from view instead of using window global.

Reply:
done